<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>A new pipeline to explore structural similarity across metabolite modules</title>
<style type="text/css">
body {
  font-family: sans-serif;
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 1.5;
  box-sizing: border-box;
}
body, .footnotes, code { font-size: .9em; }
li li { font-size: .95em; }
*, *:before, *:after {
  box-sizing: inherit;
}
pre, img { max-width: 100%; }
pre, pre:hover {
  white-space: pre-wrap;
  word-break: break-all;
}
pre code {
  display: block;
  overflow-x: auto;
}
code { font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace; }
:not(pre) > code, code[class] { background-color: #F8F8F8; }
code.language-undefined, pre > code:not([class]) {
  background-color: inherit;
  border: 1px solid #eee;
}
table {
  margin: auto;
  border-top: 1px solid #666;
}
table thead th { border-bottom: 1px solid #ddd; }
th, td { padding: 5px; }
thead, tfoot, tr:nth-child(even) { background: #eee; }
blockquote {
  color: #666;
  margin: 0;
  padding-left: 1em;
  border-left: 0.5em solid #eee;
}
hr, .footnotes::before { border: 1px dashed #ddd; }
.frontmatter { text-align: center; }
#TOC .numbered li { list-style: none; }
#TOC .numbered { padding-left: 0; }
#TOC .numbered ul { padding-left: 1em; }
table, .body h2 { border-bottom: 1px solid #666; }
.body .appendix, .appendix ~ h2 { border-bottom-style: dashed; }
.footnote-ref a::before { content: "["; }
.footnote-ref a::after { content: "]"; }
.footnotes::before {
  content: "";
  display: block;
  max-width: 20em;
}

@media print {
  body {
    font-size: 12pt;
    max-width: 100%;
  }
  tr, img { page-break-inside: avoid; }
}
@media only screen and (min-width: 992px) {
  pre { white-space: pre; }
}
</style>
</head>
<body>
<div class="include-before">
</div>
<div class="frontmatter">
<div class="title"><h1>A new pipeline to explore structural similarity across metabolite modules</h1></div>
<div class="author"><h2>| Ebtesam Abdel-Shafy, Tadele Melak, David A. MacIntyre,
| Giorgia Zadra, Luiz F. Zerbini, Silvano Piazza, and Stefano Cacciatore</h2></div>
<div class="date"><h3>2023-07-05</h3></div>
</div>
<div class="body">
<h2 id="1-introduction">1  Introduction</h2>
<p>MetChem is an R package used to perform structural and functional analysis of metabolites using a simple pipeline.</p>
<h2 id="2-installation">2	Installation</h2>
<h3 id="2-1-installation-via-cran">2.1 Installation via CRAN</h3>
<p>The R package MetChem (current version 0.2) is part of the Comprehensive R Archive Network (CRAN)^[https://cran.r-project.org/]. The simplest way to install the package is to enter the following command into your R session: <code>install.packages(&quot;MetChem&quot;)</code>. We suggest installing the following R packages: <code>pheatmap</code> and <code>RColorBrewer</code> to enable data visualization in heatmaps, <code>readxl</code> for the data reading of Excel files, and <code>impute</code> for the imputation of missing data.</p>
<pre><code># To install the pheatmap package
install.packages(&quot;pheatmap&quot;)

# To install the RColorBrewer package
install.packages(&quot;RColorBrewer&quot;)

# To install the readxl package
install.packages(&quot;readxl&quot;)

# To install the impute package
if (!require(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;impute&quot;)
</code></pre>
<h3 id="2-2-manual-installation-from-source">2.2 Manual installation from source</h3>
<p>The package can be manually installed from source by opening the package’s page in CRAN and then proceeding as follows:</p>
<ul>
<li>Download MetChem.tar.gz and save it to your hard disk</li>
<li>Open a shell/terminal/command prompt window and change to the desired directory for installation of MetChem.tar.gz. Enter <code>R CMD INSTALL MetChem.tar.gz</code> to install the package.
Note that this may require additional software on some platforms. Windows requires Rtools^[https://developer.apple.com/xcode/] to be installed and to be available in the default search path (environment variable PATH). MAC OS X requires installation of Xcode developers and command line tools.</li>
</ul>
<h3 id="2-3-compatibility-issues">2.3 Compatibility issues</h3>
<p>The package downloadable from CRAN was built using R version, R.4.2.1. The package should work without major issues on R versions &gt; 3.5.0 and KODAMA package &gt;= 2.3.</p>
<h2 id="3-getting-started">3  Getting Started</h2>
<p>The R package MetChem depends by the R package rcdk, which is partially implemented in Java. In Windows environment, we reported an issue related to the uploading of rcdk package. It can be solved running the following code before loading the package MetChem.</p>
<pre><code>replacement &lt;- function(category = &quot;LC_ALL&quot;) {
  if (identical(category, &quot;LC_MESSAGES&quot;))
    return(&quot;&quot;)
  category &lt;- match(category, .LC.categories)
  if (is.na(category)) 
    stop(&quot;invalid 'category' argument&quot;)
  .Internal(Sys.getlocale(category))

}
base &lt;- asNamespace(&quot;base&quot;)
environment(replacement) &lt;- base
unlockBinding(&quot;Sys.getlocale&quot;, base)
assign(&quot;Sys.getlocale&quot;, replacement, envir = base)
lockBinding(&quot;Sys.getlocale&quot;, base)
</code></pre>
<p>To load the package, enter the following command in your R session:</p>
<pre><code>library(&quot;MetChem&quot;)
</code></pre>
<p>If this command terminates without any error messages, the package is installed successfully. The MetChem package is now ready for use.</p>
<p>The package includes both a user manual (this document) and a reference manual (help pages for each function). To view the user manual, enter <code>vignette(&quot;MetChem&quot;)</code>. Help pages can be viewed using the command <code>help(package=&quot;MetChem&quot;)</code>.</p>
<p>\newpage</p>
<h2 id="4-example-1-murine-prostate-tissues-metabolic-profile">4  Example 1: murine prostate tissues metabolic profile</h2>
<p>Here, we introduce an example for the analysis of metabolic structural information using MetChem package. For this, we used a data set of mass spectrometry dataset obtained from murine prostate tissue samples reported by Labbé and Zadra \emph{et al.} (2019) (Supplementary Data 2). The metabolic data are obtained from ventral prostate tissues of mice that overexpress a human c-MYC transgene (MYC) in the prostate epithelium and wild-type littermates (WT). Mice were fed either a high fat diet (HFD; 60% kcal from fat; lard—rich in saturated fat) or a control diet (CTD; 10% kcal from fat). The data set includes six replicates for each group (\emph{i.e.}, WT_CTD, MYC_CTD, WT_HFD, and MYC_HFD). To begin, download the data from the Labbé and Zadra (2019) study. Download it and save it to your hard disk.
Metabolomic data is extracted using the instructions below. Data is then imputed using a k-nearest neighbour (kNN) algorithm using the function <code>impute</code> as described in the publication.</p>
<pre><code>require(&quot;readxl&quot;)
require(&quot;impute&quot;)
d=as.data.frame(read_excel(&quot;41467_2019_12298_MOESM5_ESM.xlsx&quot;,skip = 3))
d=d[1:414,]
rownames(d)=d[,&quot;Metabolite&quot;]
met=d[,4:27]
label=rep(c(&quot;WT_CTD&quot;,&quot;MYC_CTD&quot;,&quot;WT_HFD&quot;,&quot;MYC_HFD&quot;),each=6)
label_MYC=rep(c(&quot;WT&quot;,&quot;MYC&quot;,&quot;WT&quot;,&quot;MYC&quot;),each=6)
colnames(met)=paste(label,1:6)
met=data.matrix(met)
met=impute.knn(met,k=5)$data
</code></pre>
<p>Heatmap visualization is generated using the function <code>pheatmap</code>. Metabolites are hierarchically clustered according to their relative concentration The hierarchical clustering is performed using the distance matrix based on the KODAMA dimensions. KODAMA is a learning algorithm for unsupervised feature extraction specifically designed for analyzing noisy and high-dimensional data sets (Cacciatore \emph{et al.}, 2014), implemented in the R package <code>KODAMA</code> (Cacciatore \emph{et al.}, 2017). Additional information can be found in the review of Zinga \emph{et al.}, 2023.</p>
<pre><code>require(&quot;pheatmap&quot;)
require(&quot;RColorBrewer&quot;)

my_colour1 = list(genotype=c(MYC=&quot;#000000ff&quot;,WT=&quot;#eeeeeeff&quot;),
                  group=c(MYC_CTD=&quot;#373898ff&quot;,MYC_HFD=&quot;#c11630ff&quot;,
                          WT_CTD=&quot;#00a4cfff&quot;,WT_HFD=&quot;#e40a81ff&quot;))

set.seed(1)
kk1=KODAMA.matrix(t(met))
col=KODAMA.visualization(kk1)
hcol=hclust(dist(col),method=&quot;ward.D&quot;)

kk2=KODAMA.matrix(scale(met))
row=KODAMA.visualization(kk2)
hrow=hclust(dist(row),method=&quot;ward.D&quot;)
my_sample_col &lt;- data.frame(group = label,genotype=label_MYC)
row.names(my_sample_col) &lt;- colnames(met)

pheatmap(met,
         cluster_cols = hcol,
         cluster_rows = hrow,
         labels_row = rep(&quot;&quot;,nrow(met)),
         annotation_col = my_sample_col, 
         annotation_colors = my_colour1,
         color = colorRampPalette(rev(brewer.pal(n = 11, name =&quot;RdBu&quot;)))(100))
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=12cm]{KODAMApaper.pdf}}
\caption{Heatmap of metabolites with hierarchical clustering based on their concentratrion.}
\label{fig:F1}
\end{figure}</p>
<p>\newpage</p>
<p>To analyze the chemical similarities among metabolites, we need the Simplified Molecular-Input Line-Entry System (SMILES) of each metabolite is obtained. The SMILES of the previous data set is stored in the list <code>HFD</code> that can be loaded using the function <code>data(HFD)</code>. <code>MetChem</code> package includes the <code>modules.detection</code> function based on KODAMA analysis.  This function repeats the follwing steps (10 times as default): i) transformation of the chemical structure dissimilarity matrix in a multidimensional space (with 50 dimensions as defaults) using multidimensional scaling; ii) KODAMA features extraction; iii) hierarchical clustering based on the KODAMA output; iv) Calculation of the silhoutte index from different number of clusters (from 2 to 30 as default). The average of the siloutte index is calculated for each cluster numbers to identity the optimal cluster number.</p>
<pre><code>data(HFD)
met=met[rownames(HFD),]
SMILES=HFD$SMILES
names(SMILES)=rownames(HFD)
clu=modules.detection(SMILES)
plot(clu$min_nc:clu$max_nc,clu$silhouette,type=&quot;l&quot;,
     ylab=&quot;Rousseeuw's Silhouette index&quot;,xlab=&quot;Number of clusters&quot;)
abline(v=5*(1:6),lty=2)

print(clu$main_cluster)
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=8cm]{FLindex.pdf}}
\caption{Silhouette index.}
\label{fig:F2}
\end{figure}</p>
<p>\newpage</p>
<p>Based on the average Silhouette indeces, we have the optimal number of cluster for this data set is 8. Below is shown a graphical visualization of the final output of KODAMA. Each cluster is represented by a different color code. Each dot represents a different metabolite. Metabolites that are located near to each other share a similar chemical structure.</p>
<pre><code>plot(clu$visualization,pch=21,bg=rainbow(8,alpha = 0.7)[clu$clusters[,&quot;Clusters 8&quot;]],cex=2)
legend(-30, 20, legend=paste(&quot;Cluster&quot;, unique(clu$clusters[,&quot;Clusters 8&quot;])),
   col= rainbow(8,alpha = 0.7), pch= 16, cex=1)
</code></pre>
<p>The following line code can be used to identify the points on the scatter plot. The cluster belonging and chemical name of the selected points will be displayed. ESC key to terminate the command.</p>
<pre><code>data.frame(metabolite=rownames(met),
           cluster=clu$clusters[,&quot;Clusters 8&quot;])[identify(clu$visualization),]
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=12cm]{KODAMA.pdf}}
\caption{KODAMA plot.}
\label{fig:F3}
\end{figure}</p>
<p>\newpage</p>
<p>A new heatmap is generated where metabolites are clustered according to their chemical similarity.</p>
<pre><code>my_colour2 = list(cluster = c(&quot;1&quot;=rainbow(8,alpha = 1)[1],
                              &quot;2&quot;=rainbow(8,alpha = 1)[2],
                              &quot;3&quot;=rainbow(8,alpha = 1)[3],
                              &quot;4&quot;=rainbow(8,alpha = 1)[4],
                              &quot;5&quot;=rainbow(8,alpha = 1)[5],
                              &quot;6&quot;=rainbow(8,alpha = 1)[6],
                              &quot;7&quot;=rainbow(8,alpha = 1)[7],
                              &quot;8&quot;=rainbow(8,alpha = 1)[8]),
                  genotype=c(MYC=&quot;#000000ff&quot;,WT=&quot;#eeeeeeff&quot;),
                  group=c(MYC_CTD=&quot;#373898ff&quot;,MYC_HFD=&quot;#c11630ff&quot;,
                          WT_CTD=&quot;#00a4cfff&quot;,WT_HFD=&quot;#e40a81ff&quot;))

clusters8=clu$clusters[,&quot;Clusters 8&quot;]
my_sample_row &lt;- data.frame(cluster = as.character(clusters8))
row.names(my_sample_row) &lt;- rownames(met)

set.seed(1)
met=met[rownames(HFD),] 
kk1=KODAMA.matrix(t(met))
col=KODAMA.visualization(kk1)
hcol=hclust(dist(col),method=&quot;ward.D&quot;)
hrow=clu$hclust

my_sample_col &lt;- data.frame(group = label,genotype=label_MYC)
row.names(my_sample_col) &lt;- colnames(met)

pheatmap(met,
         cluster_cols = hcol,
         cluster_rows = hrow,
         labels_row = rep(&quot;&quot;,nrow(met)), 
         annotation_colors = my_colour2,
         annotation_col = my_sample_col, 
         annotation_row = my_sample_row, 
         cutree_rows = 8, 
         color = colorRampPalette(rev(brewer.pal(n = 11, name =&quot;RdBu&quot;)))(100))
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=12cm]{heatmap.pdf}}
\caption{Heatmap of metabolites with vertical hierarchical clustering based on their molecular structure.}
\label{fig:F4}
\end{figure}</p>
<p>\newpage</p>
<p>We next build a heatmap of the metabolite belonging to the cluster 4.</p>
<pre><code>sel=clu$clusters[,&quot;Clusters 8&quot;]==4
met.sel=met[sel,] 
my_sample_col &lt;- data.frame(group = label,genotype=label_MYC)
row.names(my_sample_col) &lt;- colnames(met)
oo=order(row.names(my_sample_col))
my_sample_col=my_sample_col[oo,]
met.sel=met.sel[,oo]
hrow.sel=hclust(dist(clu$visualization[sel,]),method=&quot;ward.D&quot;) 

pheatmap(met.sel,fontsize = 7,
         cluster_cols = FALSE,
         cluster_rows = hrow.sel,
         annotation_col = my_sample_col, 
         annotation_colors = my_colour1,
         color = colorRampPalette(rev(brewer.pal(n = 11, name =&quot;RdBu&quot;)))(100))
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=12cm]{KODAMAselection.pdf}}
\caption{Heatmap}
\label{fig:F6}
\end{figure}</p>
<p>\newpage</p>
<p>In the next step, we apply the Weighted Metabolite Chemical Similarity Analysis (WMCSA) on all branches of the hierarchical clustering performed on the cluster 4. WMCSA is implemented in the function <code>WMCSA</code>. This function summarizes the relative concentration of metabolites within each module. Each module is defined according to the chemical similarity.</p>
<pre><code>set.seed(1)
my_sample_col &lt;- data.frame(group = label,genotype=label_MYC)
row.names(my_sample_col) &lt;- colnames(met)
cl=allbranches(hrow.sel)
ww=WMCSA(met.sel,cl)

hrow=hclust(dist(ww),method=&quot;ward.D&quot;)

pheatmap(ww,
         cluster_cols = FALSE,
         cluster_rows = hrow,
         annotation_col = my_sample_col, 
         annotation_colors = my_colour1,
         color = colorRampPalette(rev(brewer.pal(n = 11, name =&quot;RdBu&quot;)))(100))
</code></pre>
<p>\begin{figure}[h]
\centerline{\includegraphics[width=12cm]{KODAMAall.pdf}}
\caption{Heatmap of the output of WMCSA.}
\label{fig:F5}
\end{figure}</p>
<p>\newpage</p>
<p>Differential analysis of the relevant modules can be performed using the function <code>multi_analysis</code> present in the R package <code>KODAMA</code>. In the example below, we perform a differential analysis between MYC transgenic mice fed with high-fat diet or control diet named as MYC_HFD and MYC_CTD, respectively.</p>
<pre><code>label=rep(c(&quot;MYC&quot;,&quot;WT&quot;),each=12)
multi_analysis(t(ww),label)
</code></pre>
<table>
<thead>
<tr>
<th align="left">Feature</th>
<th align="center">MYC</th>
<th align="center">WT</th>
<th align="center">p-value</th>
<th align="center">FDR</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Mod1, median [IQR]</td>
<td align="center">-2.919 [-3.833 -1.536]</td>
<td align="center">2.938 [2.195 3.563]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod2, median [IQR]</td>
<td align="center">-2.17 [-2.638 -1.706]</td>
<td align="center">2.237 [1.98 2.478]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod3, median [IQR]</td>
<td align="center">-1.798 [-2.033 -1.215]</td>
<td align="center">1.409 [1.015 2.228]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod4, median [IQR]</td>
<td align="center">-2.45 [-3.037 -1.381]</td>
<td align="center">2.139 [1.776 2.827]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod5, median [IQR]</td>
<td align="center">-1.902 [-2.171 -1.425]</td>
<td align="center">1.688 [1.563 1.941]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod6, median [IQR]</td>
<td align="center">-1.772 [-3.022 -1.119]</td>
<td align="center">1.701 [0.742 2.204]</td>
<td align="center">2.46e-04</td>
<td align="center">4.10e-04</td>
</tr>
<tr>
<td align="left">Mod7, median [IQR]</td>
<td align="center">0.886 [-1.181 2.231]</td>
<td align="center">-0.329 [-1.224 0.379]</td>
<td align="center">4.03e-01</td>
<td align="center">4.64e-01</td>
</tr>
<tr>
<td align="left">Mod8, median [IQR]</td>
<td align="center">-1.674 [-2.485 -0.656]</td>
<td align="center">1.248 [0.609 1.995]</td>
<td align="center">3.84e-04</td>
<td align="center">5.76e-04</td>
</tr>
<tr>
<td align="left">Mod9, median [IQR]</td>
<td align="center">1.609 [0.961 2.06]</td>
<td align="center">-1.412 [-1.706 -1.203]</td>
<td align="center">4.69e-05</td>
<td align="center">8.80e-05</td>
</tr>
<tr>
<td align="left">Mod10, median [IQR]</td>
<td align="center">-0.84 [-1.956 0.087]</td>
<td align="center">0.8 [-0.148 1.161]</td>
<td align="center">4.64e-02</td>
<td align="center">6.33e-02</td>
</tr>
<tr>
<td align="left">Mod11, median [IQR]</td>
<td align="center">-0.023 [-0.639 1.15]</td>
<td align="center">-0.583 [-1.229 0.027]</td>
<td align="center">1.75e-01</td>
<td align="center">2.19e-01</td>
</tr>
<tr>
<td align="left">Mod12, median [IQR]</td>
<td align="center">-1.347 [-1.928 -0.681]</td>
<td align="center">1.305 [0.711 1.736]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod13, median [IQR]</td>
<td align="center">0.757 [-0.717 1.532]</td>
<td align="center">-0.004 [-1.003 0.6]</td>
<td align="center">5.07e-01</td>
<td align="center">5.43e-01</td>
</tr>
<tr>
<td align="left">Mod14, median [IQR]</td>
<td align="center">1.691 [0.59 1.952]</td>
<td align="center">-1.389 [-1.662 -0.889]</td>
<td align="center">3.66e-05</td>
<td align="center">7.84e-05</td>
</tr>
<tr>
<td align="left">Mod15, median [IQR]</td>
<td align="center">0.196 [-1.074 0.842]</td>
<td align="center">0.388 [-0.853 0.638]</td>
<td align="center">7.95e-01</td>
<td align="center">7.95e-01</td>
</tr>
</tbody>
</table>
<p>\newpage</p>
<p>\newpage
The function <code>readMet</code> connects R to the HMDB database ^[https://www.hmdb.ca]. to retrieve chemical and functional information of each metabolite. This can be summarized using different functions: <code>substituentsMet</code>, <code>diseaseMet</code>, <code>enzymeMet</code>, <code>pathwaysMet</code>, <code>taxonomyMet</code>. The function <code>features</code> associates the most prominent features to each module.</p>
<p>In this example, we characterized the modules by functional group.</p>
<pre><code>HMDB=HFD$HMDB
names(HMDB)=rownames(HFD)
doc=readMet(HMDB)
cla1=substituentsMet(doc)
substituents=features(doc,cla1,cl)
cla2=enzymesMet(doc)
enzymes=features(doc,cla2,cl)
</code></pre>
<p>To visualize the metabolite of , for example, module 12, we can type:</p>
<pre><code>cl[[12]]
</code></pre>
<p>Fisher’s test was used to rank the association of each module to the metabolite information. Below are reported the p-value for associations of substituents with the module 12.</p>
<pre><code>substituents[[12]]
</code></pre>
<table>
<thead>
<tr>
<th align="left">Substituents</th>
<th align="center">p-value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Histidine or derivatives</td>
<td align="center">1.44e-06</td>
</tr>
<tr>
<td align="left">Imidazolyl carboxylic acid derivative</td>
<td align="center">1.21e-05</td>
</tr>
<tr>
<td align="left">Hybrid peptide</td>
<td align="center">4.97e-05</td>
</tr>
<tr>
<td align="left">N-acyl-alpha amino acid or derivatives</td>
<td align="center">1.21e-03</td>
</tr>
<tr>
<td align="left">Imidazole</td>
<td align="center">1.31e-03</td>
</tr>
<tr>
<td align="left">Azole</td>
<td align="center">1.53e-03</td>
</tr>
<tr>
<td align="left">Aromatic heteromonocyclic compound</td>
<td align="center">1.77e-03</td>
</tr>
<tr>
<td align="left">N-acyl-alpha-amino acid</td>
<td align="center">2.33e-03</td>
</tr>
<tr>
<td align="left">Beta amino acid or derivatives</td>
<td align="center">8.62e-03</td>
</tr>
<tr>
<td align="left">Organoheterocyclic compound</td>
<td align="center">1.18e-02</td>
</tr>
<tr>
<td align="left">Heteroaromatic compound</td>
<td align="center">1.47e-02</td>
</tr>
<tr>
<td align="left">Carboximidic acid</td>
<td align="center">1.91e-02</td>
</tr>
<tr>
<td align="left">Carboximidic acid derivative</td>
<td align="center">2.04e-02</td>
</tr>
<tr>
<td align="left">Azacycle</td>
<td align="center">2.05e-02</td>
</tr>
<tr>
<td align="left">Organic 1,3-dipolar compound</td>
<td align="center">2.32e-02</td>
</tr>
<tr>
<td align="left">Propargyl-type 1,3-dipolar organic compound</td>
<td align="center">2.32e-02</td>
</tr>
<tr>
<td align="left">Alpha-amino acid or derivatives</td>
<td align="center">3.95e-02</td>
</tr>
<tr>
<td align="left">Amino acid or derivatives</td>
<td align="center">4.13e-02</td>
</tr>
<tr>
<td align="left">\newpage</td>
<td align="center"></td>
</tr>
</tbody>
</table>
<p>Here  are reported the p-value for associations of metabolite-related enzymes with the module 12.</p>
<pre><code>enzymes[[12]]
</code></pre>
<table>
<thead>
<tr>
<th align="left">Substituents</th>
<th align="center">p-value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">CNDP1</td>
<td align="center">2.97e-04</td>
</tr>
<tr>
<td align="left">CARNS1</td>
<td align="center">1.38e-03</td>
</tr>
<tr>
<td align="left">VIM</td>
<td align="center">8.62e-03</td>
</tr>
<tr>
<td align="left">HSPA1A</td>
<td align="center">8.62e-03</td>
</tr>
<tr>
<td align="left">SLC15A2</td>
<td align="center">8.62e-03</td>
</tr>
<tr>
<td align="left">MPO</td>
<td align="center">2.57e-02</td>
</tr>
</tbody>
</table>
<h2 id="4-how-to-cite-this-package">4  How to Cite this Package</h2>
<p>Ebtesam Abdel-Shafy, Tadele Melak, David A. MacIntyre, Giorgia Zadra, Luiz F. Zerbini, Silvano Piazza, and Stefano Cacciatore
Publication in submission</p>
<p>To obtain BibTex entries of the two references, you can enter the following into your R session to Bibtex <code>citation(&quot;MetChem&quot;)</code>.</p>
<h2 id="5-references">5  References</h2>
<p>Cacciatore S, Luchinat C, Tenori L. Knowledge discovery by accuracy maximization. \emph{Proc Natl Acad Sci USA} 2014; 111: 5117-22.</p>
<p>Cacciatore S, Tenori L, Luchinat C, Bennett PR, MacIntyre DA (2017) KODAMA: an R package for knowledge discovery and data mining. \emph{Bioinformatics} 2017; 33(4): 621-623.</p>
<p>Labbé DP, Zadra G, Yang M, Reyes JM, Lin CY, Cacciatore S, Ebot EM, Creech AL, Giunchi F, Fiorentino M, Elfandy H, Syamala S, Karoly ED, Alshalalfa M, Erho N, Ross A, Schaeffer EM, Gibb EA, Takhar M, Den RB, Lehrer J, Karnes RJ, Freedland SJ, Davicioni E, Spratt DE, Ellis L, Jaffe JD, D’Amico AV, Kantoff PW, Bradner JE, Mucci LA, Chavarro JE, Loda M, Brown M. High-fat diet fuels prostate cancer progression by rewiring the metabolome and amplifying the MYC program. \emph{Nat Commun} 2019; 10: 4358.</p>
<p>Zinga MM, Abdel-Shafy E, Melak T, Vignoli A, Piazza S, Zerbini LF, Tenori L, Cacciatore S. KODAMA exploratory analysis in metabolic phenotyping. \emph{Front Mol Biosci} 2023; 9: 1436.</p>
</div>
<div class="include-after">
</div>
</body>
</html>
